lib/ClangImporter/cxx-interop.h

```diff
@@ -0,0 +1,31 @@
+//===--- cxx-interop.h - C++ and Swift Interop ------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file provides common utilities and annotations that are useful for C++
+// codebases that interoperate with Swift.
+//
+//===----------------------------------------------------------------------===//
+#ifndef SWIFT_CLANGIMPORTER_CXX_INTEROP_H
+#define SWIFT_CLANGIMPORTER_CXX_INTEROP_H
+
+#define SELF_CONTAINED __attribute__((swift_attr("import_owned")))
+#define SAFE_TO_IMPORT __attribute__((swift_attr("import_unsafe")))
+
+#define _CXX_INTEROP_STRINGIFY(_x) #_x
+#define REFERENCE(_retain, _release)                                           \
+  __attribute__((swift_attr("import_reference")))                              \
+  __attribute__((swift_attr(_CXX_INTEROP_STRINGIFY(retain:_retain))))          \
+  __attribute__((swift_attr(_CXX_INTEROP_STRINGIFY(release:_release))))
+
+#define SWIFT_NAME(_name) __attribute__((swift_name(#_name)))
```

Comment:
I don't think `SWIFT_NAME` belongs here: it isn't specific to C++ interop, it is also used for C/ObjC interop. On Darwin there's an existing macro for this (`NS_SWIFT_NAME`), is this a replacement for `NS_SWIFT_NAME`?

Reply:
I was thinking more about non-darwin platforms or users that don't have access to or don't want to use Foundation (the place this macro is currently provided).